indigo_app/views/tasks.py

```diff
@@ -327,13 +336,25 @@ def post(self, request, *args, **kwargs):
 
         task.save()
 
+        if self.change == 'close' or self.change == 'cancel':
+            self.update_blocked_tasks(task, user)
+
         return redirect(self.get_redirect_url())
 
     def get_redirect_url(self):
         if self.request.GET.get('next'):
             return self.request.GET.get('next')
         return reverse('task_detail', kwargs={'place': self.kwargs['place'], 'pk': self.kwargs['pk']})
 
+    def update_blocked_tasks(self, task, user):
+        previously_blocked_tasks = list(task.blocking.all())
+        # this task is no longer blocking other tasks
+        task.blocking.clear()
+        for blocked_task in previously_blocked_tasks:
+            if not blocked_task.blocked_by.all() and has_transition_perm(blocked_task.unblock, self):
```

Comment:
use `.exist()` rather than loading all the data from the db and not using it.

should `self` at the end be `user`?

Reply:
I was thinking of refactoring the may_{verb} code, but ya they all use the view, not the user